measuring-client/src/main/java/de/cyface/app/ui/button/DataCapturingButton.java

```diff
@@ -901,50 +912,50 @@ public void onFixLost() {
+            final String minutesText = durationMinutes > 0 ? durationMinutes % 60 + "m " : "";
+            final String secondsText = durationSeconds % 60 + "s";
+            final String durationText = hoursText + minutesText + secondsText;
+            Log.d(TAG, "Duration update: " + durationMillis);
+            durationTextView.setText(durationText);
+
+            final double averageSpeedKmh = persistenceLayer.loadAverageSpeed(measurement.getIdentifier(),
+                    new DefaultLocationCleaningStrategy()) * 3.6;
+            final String averageSpeedText = Math.round(averageSpeedKmh) + " km/h";
+            Log.d(TAG, "Average speed update: " + averageSpeedText);
+            averageSpeedTextView.setText(averageSpeedText);
+
+            addLocationToCachedTrack(geoLocation);
+            final List<Event> currentMeasurementsEvents = loadCurrentMeasurementsEvents();
+            mainFragment.getMap().renderMeasurement(currentMeasurementsTracks, currentMeasurementsEvents, false);
+        } catch (final CursorIsNullException e) {
+            Log.w(TAG, "onNewGeoLocationAcquired(): Cursor is null, doing nothing.");
             if (!onNewGeoLocationAcquiredExceptionTriggered[0] && isReportingEnabled) {
                 onNewGeoLocationAcquiredExceptionTriggered[0] = true;
                 Sentry.captureException(e);
             }
-            return;
-        } catch (final CursorIsNullException e) {
-            throw new IllegalStateException(e);
-        }
-        final int distanceMeter = (int)Math.round(measurement.getDistance());
-        final double distanceKm = distanceMeter == 0 ? 0.0 : distanceMeter / 1000.0;
-        final String distanceText = distanceKm + " km";
-        Log.d(TAG, "Distance update: " + distanceText);
-        distanceTextView.setText(distanceText);
-        Log.d(TAG, "distanceTextView: " + distanceTextView.getText());
-        Log.d(TAG, "distanceTextView: " + distanceTextView.isEnabled());
-
-        addLocationToCachedTrack(geoLocation);
-        final List<Event> currentMeasurementsEvents;
-        try {
-            currentMeasurementsEvents = loadCurrentMeasurementsEvents();
-            mainFragment.getMap().renderMeasurement(currentMeasurementsTracks, currentMeasurementsEvents, false);
-        } catch (final CursorIsNullException e) {
-            Log.w(TAG, "onNewGeoLocationAcquired() failed to loadCurrentMeasurementsEvents(). "
-                    + "Thus, map.renderMeasurement() is ignored. This should only happen id "
-                    + "the capturing already stopped.");
+        } catch (NoSuchMeasurementException e) {
+            // GeoLocations may also arrive shortly after a measurement was stopped. Thus, this may not crash.
+            // This happened on the Emulator with emulated live locations.
+            Log.w(TAG, "onNewGeoLocationAcquired(): No currently captured measurement found, doing nothing.");
```

Comment:
unchanged, I just merged the try-catch sections in this method or else I would have needed 3 sections (duplicate code)